### PR TITLE
Add support for Azure Linux

### PR DESCRIPTION
**Pull request by DaanDeMeyer**
Azure Linux looks a lot like Fedora Linux so we opt to share configuration
between Azure and Fedora/CentOS and inherit the Azure definition from
Fedora.

This should not be merged until Azure Linux 3.0 moves out of preview since the mirror urls will have to change when that happens.

### CODE REVIEW

File: mkosi.conf.d/30-azure-centos-fedora/mkosi.conf.d/20-x86-64.conf [hunk 1]
```diff
@@ -7,5 +7,5 @@ Architecture=x86-64
 Packages=
         grub2-efi-x64-modules
         grub2-pc
-        kernel-uki-virt
+        # kernel-uki-virt
```
Comment (bluca) [inline mkosi.conf.d/30-azure-centos-fedora/mkosi.conf.d/20-x86-64.conf:10]: why is this commented out?

Comment (DaanDeMeyer) [reply mkosi.conf.d/30-azure-centos-fedora/mkosi.conf.d/20-x86-64.conf:10]: oops leftover

**Review by behrmann — APPROVED** [review]
Would it be possible to add it to CI in some way?

**Review by keszybz — APPROVED** [review]
LGTM too.

**Comment by bluca** [comment]
The kconfigs we need have been added: https://github.com/microsoft/azurelinux/pull/9971
not sure when the next RC will happen, but these should be included

**Review by bluca — APPROVED** [review]

**Pull request merged**